src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -306,6 +335,101 @@ .maxstack  4
 }
 ");
         }
+        [Fact]
+        public void TupleTypeDeclaration()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, string, int) x = (1, ""hello"", 2);
+        System.Console.WriteLine(x.ToString());
+    }
+}
+
+" + trivial3uple;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"{1, hello, 2}");
+        }
+
+        [Fact]
+        public void TupleTypeMismatch()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, string) x = (1, ""hello"", 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            // PROTOTYPE there should be no mention of Item1, Item2 or Item3 in there error for this test
+            CreateCompilationWithMscorlib(source).VerifyDiagnostics(
+                // (6,27): error CS0029: Cannot implicitly convert type '<tuple: int Item1, string Item2, int Item3>' to '<tuple: int Item1, string Item2>'
+                //         (int, string) x = (1, "hello", 2);
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"(1, ""hello"", 2)").WithArguments("<tuple: int Item1, string Item2, int Item3>", "<tuple: int Item1, string Item2>").WithLocation(6, 27));
+        }
+
+        [Fact]
+        public void TupleTypeWithLateDiscoveredName()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, string a) x = (1, ""hello"", c: 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            var tree = Parse(source);
+            var comp = CreateCompilationWithMscorlib(tree);
```

Comment:
The diagnostic is that all the members should have names. That's already covered in previous test. 
I'm leaning to not duplicate. What do you think?